### PR TITLE
fix for python3 TypeError: ord() expected string of length 1, but int…

### PR DESCRIPTION
**Pull request by slieberth**
python3 compatibility for SaltCrypt()

### CODE REVIEW

File: pyrad/packet.py [hunk 1]
```diff
@@ -563,15 +563,27 @@ def SaltCrypt(self, value):
             # self.authenticator = self.CreateAuthenticator()
             self.authenticator = 16 * six.b('\x00')
 
-        salt = struct.pack('!H', random_generator.randrange(0, 65535))
-        salt = chr(ord(salt[0]) | 1 << 7)+salt[1]
+        if six.PY3:
+            random_value = 32768 + random_generator.randrange(0, 32767)
```
Comment (Istvan91) [inline pyrad/packet.py:567]: I don't think this is correct, this is effectively a random number between 32768 and 65535 when it should be between 0 and 65535 (the full range of 2 bytes). The random value calculation is the same for the python2 and python3 case, so move it please out of the conditional blocks.

File: pyrad/packet.py [hunk 2]
```diff
@@ -563,15 +563,27 @@ def SaltCrypt(self, value):
...
+        #salt = struct.pack('!H', random_generator.randrange(0, 65535))
+        #salt = chr(ord(salt[0]) | 1 << 7)+salt[1]
 
         length = struct.pack("B", len(value))
         buf = length + value
         if len(buf) % 16 != 0:
             buf += six.b('\x00') * (16 - (len(buf) % 16))
 
-        result = six.b(salt)
+        #result = six.b(salt)
```
Comment (Istvan91) [inline pyrad/packet.py:586]: Please delete commented out statements.

**Comment by slieberth** [comment]
Hi Istvan, please double check RFC 2868:
   Salt
      The Salt field is two octets in length and is used to ensure the
      uniqueness of the encryption key used to encrypt each instance of
      the Tunnel-Password attribute occurring in a given Access-Accept
      packet.  **The most significant bit (leftmost) of the Salt field
      MUST be set (1).**  The contents of each Salt field in a given
      Access-Accept packet MUST be unique.

**Comment by Istvan91** [comment]
thanks for pointing that out, I've looked at the RFC earlier, but I completely missed that.
Nevertheless fixing it for the "old" python2 case would nice.

**Pull request merged**